packages/plugin-kafka/src/custom/consumer-stream.ts

```diff
@@ -200,11 +199,24 @@ export class KafkaConsumerStream extends Readable {
   }
 
   public close(cb?: (err?: Error | null, result?: any) => void): void {
-    this.consumer.disconnect(cb)
+    this.consumer.disconnect()
+
+    if (this.endEmitted || !this.consumer.isConnected()) {
+      if (cb) cb()
+      return
+    }
+
+    this.on('error', (err) => {
+      if (cb) cb(err)
+    })
+
+    this.on('close', () => {
+      if (cb) cb()
+    })
```

Comment:
event handles wont get cleaned up, needs to be once or ideally be the same as it was before